src/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,318 @@
+        debtAmount += amount;
+
+        if (debtAmount > borrowCapacity) {
+            revert InsufficentCollateral(borrowCapacity, debtAmount);
+        }
+
+        positions[msg.sender].debtAmount = debtAmount;
+        
+        // If more than 1 interest rate period has passed update the start-time
+        if (block.timestamp - positions[msg.sender].createdAt >= interestRatePeriod || positions[msg.sender].createdAt == 0 ) {
+            positions[msg.sender].createdAt = block.timestamp;
+        }
+         
+        debtBalance -= amount;
+        debtToken.safeTransfer(
+            msg.sender,
+            amount
+        );
+        emit Borrow(msg.sender, amount);
+    }
+
+    function maxBorrowCapacity(address account) public view returns(uint256) {
+        return ((positions[account].collateralAmount * collateralPrice * 100) / debtPrice / minCollateralizationRatio);
+    }
+
+
+    function _maxBorrowCapacity(uint256 collateralAmount) internal view returns (uint256) {
+        return collateralAmount * collateralPrice * 100 / debtPrice / minCollateralizationRatio;
+    }
+
+   /**
+     * @dev Allows borrower to with draw collateral if sufficient to not default on loan
+     * @param withdrawalAmount is the amount to withdraw
+     */
+    function withdrawCollateral(uint256 withdrawalAmount) external {
+
+        uint256 collateralAmount = positions[msg.sender].collateralAmount;
+        if (withdrawalAmount > collateralAmount) {
+            revert ExceededCollateralAmonut(msg.sender, collateralAmount, withdrawalAmount);
+        }
+
+        uint256 borrowCapacity = _maxBorrowCapacity(collateralAmount - withdrawalAmount);
+        
+        if (positions[msg.sender].debtAmount > borrowCapacity ) {
+            revert WillUnderCollaterlize(msg.sender, withdrawalAmount);
+        }
+
+        positions[msg.sender].collateralAmount -= withdrawalAmount;
+        collateralToken.safeTransfer(
+            msg.sender,
+            withdrawalAmount
+        );
+
+        emit Withdraw(msg.sender, withdrawalAmount);
+    }
+
+   /**
+     * @dev Allows borrower to repay borrowed amount
+     * @param repayAmount is the amount to repay
+     */
+    function repay(uint256 repayAmount) external {
+        if (repayAmount == 0) revert InvalidAmount(repayAmount);
+        positions[msg.sender].debtAmount = getTotalDebtAmount(msg.sender);
+
+        if (repayAmount >  positions[msg.sender].debtAmount) {
+            revert ExceededBorrowedAmount(msg.sender, positions[msg.sender].debtAmount, repayAmount);
+        }
+
+        positions[msg.sender].debtAmount -= repayAmount;
+        debtBalance += repayAmount;
+        
+        // If more than 1 interest rate period has passed update the start-time
+        if (block.timestamp - positions[msg.sender].createdAt >= interestRatePeriod || positions[msg.sender].createdAt == 0  ) {
+            positions[msg.sender].createdAt = block.timestamp;
```

Comment:
I don't think the if() is required here or in borrow(), if there's no truncation in the getTotalDebtAmount() function. So can just always set:

```
positions[msg.sender].createdAt = block.timestamp;
```

Definitely worth tests which check the edge cases carefully